apps/admin-gui/src/app/vos/components/group-add-member-dialog/group-add-member-dialog.component.ts

```diff
@@ -89,144 +91,74 @@ export class GroupAddMemberDialogComponent implements OnInit {
           ]);
       }
     });
-  }
-
-  add(): void {
-    this.loading = true;
-    if (this.selection.selected.length === 0) {
-      if (this.failed.length !== 0) {
-        this.failed = this.failed.filter((failed) => failed !== null);
-        this.selection.clear();
-        this.loading = false;
-      } else {
-        this.addMemberService.success('DIALOGS.ADD_MEMBERS.SUCCESS_ADD');
-      }
-      return;
-    }
 
-    this.requestService.dontHandleErrorForNext();
-    const candidate = this.selection.selected.pop();
-
-    if (candidate.member) {
-      this.addMember(candidate);
-    } else if (candidate.richUser) {
-      this.addUser(candidate);
-    } else if (candidate.candidate) {
-      this.addCandidate(candidate);
-    }
+    this.registrarManager
+      .invitationFormExists(this.data.group.voId, this.data.group.id)
+      .subscribe((res) => {
+        this.showInvite = res;
+      });
   }
 
-  invite(lang: string): void {
+  add(): void {
     this.loading = true;
-    if (this.selection.selected.length === 0) {
-      if (this.failed.length !== 0) {
-        this.failed = this.failed.filter((failed) => failed !== null);
-        this.selection.clear();
-        this.loading = false;
-      } else {
-        this.addMemberService.success('DIALOGS.ADD_MEMBERS.SUCCESS_INVITE');
-      }
-      return;
-    }
-
-    this.requestService.dontHandleErrorForNext();
-    const candidate = this.selection.selected.pop();
-
-    if (candidate.richUser) {
-      this.inviteUser(candidate, lang);
-    } else {
-      this.inviteCandidate(candidate, lang);
-    }
-  }
 
-  cancel(result: boolean): void {
-    this.dialogRef.close(result);
-  }
+    const selectedMemberCandidates = this.selection.selected.map(
+      (elem) =>
+        ({
+          member: elem.member,
+          candidate: elem.candidate,
+          richUser: elem.richUser,
+        } as MemberCandidate)
+    );
```

Comment:
Same goes for invite. It would be nice to put this logic into a separate method so change have to be done only once.
Also valid for the vo dialog.

```suggestion
    const selectedMemberCandidates: MemberCandidate[] = this.selection.selected.map((elem) => ({
      member: elem.member,
      candidate: elem.candidate,
      richUser: elem.richUser,
    }));
    );
```